### PR TITLE
Change the behaviour of Variable.__len__

### PR DESCRIPTION
**Pull request by mitmul**
Change for https://github.com/pfnet/chainer/issues/1792

### CODE REVIEW

**Comment by beam2d** [comment]
Would you fix the docstring as well?

**Comment by mitmul** [comment]
Yes, thanks for pointing out. I followed the numpy's condition for exception https://github.com/numpy/numpy/blob/master/numpy/core/src/multiarray/mapping.c#L51

File: chainer/variable.py [hunk 1]
```diff
@@ -159,13 +159,15 @@ def __pos__(self):
...
 
         Returns:
-            int: Number of elements of the data array.
+            int: Number of the first dimension of the data array.
 
         """
-        return self.data.size
+        if self.data.ndim == 0:
+            raise TypeError('len() of unsized object')
+        return self.data.shape[0]
```
Comment (beam2d) [inline chainer/variable.py]: Is it enough to just call `len(self.data)`?

**Comment by beam2d** [comment]
LGTM

**Pull request merged**